### PR TITLE
Remove a number of unused java elements.

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Overview
Remove a good number of the elements identified as unused by IDEA static analysis.

## Functional Changes
- none

### CODE REVIEW

File: game-core/src/main/java/tools/map/making/MapProperties.java [hunk 1]
```diff
@@ -3,51 +3,20 @@
...
 import games.strategy.engine.data.properties.PropertiesUi;
 import games.strategy.triplea.Constants;
-import games.strategy.triplea.image.UnitImageFactory;
-import games.strategy.triplea.ui.mapdata.MapData;
 import games.strategy.util.Tuple;
 
 /**
  * An object to hold all the map.properties values.
  */
-public class MapProperties {
```
Comment (ssoloff) [inline game-core/src/main/java/tools/map/making/MapProperties.java]: Pretty sure the bulk of this class is accessed reflectively by the Map Properties Maker tool.

File: game-core/src/main/java/swinglib/JLabelBuilder.java [hunk 2]
```diff
@@ -50,15 +48,7 @@ public static JLabelBuilder builder() {
...
-    final String truncated;
-    if (text != null) {
-      Preconditions.checkState(!text.trim().isEmpty());
-      truncated = Ascii.truncate(text, maxTextLength, "...");
-    } else {
-      truncated = "";
-    }
-
-    final JLabel label = new JLabel(truncated);
+    final JLabel label = new JLabel(text);
```
Comment (RoiEXLab) [inline game-core/src/main/java/swinglib/JLabelBuilder.java]: I'm curious to know what lead to this change in behaviour.

File: game-core/src/main/java/tools/map/making/MapPropertiesMaker.java [hunk 3]
```diff
@@ -398,27 +398,5 @@ private void handleCommandLineArgs(final String[] args) {
...
-    if (heightString != null && heightString.length() > 0) {
-      try {
-        final int unitHeight = Integer.parseInt(heightString);
-        mapProperties.setUnitsHeight(unitHeight);
-        mapProperties.setUnitsCounterOffsetHeight(unitHeight);
-        log.info("Unit Height to use: " + unitHeight);
-      } catch (final Exception e) {
-        log.severe("Not an integer: " + heightString);
-      }
-    }
```
Comment (RoiEXLab) [inline game-core/src/main/java/tools/map/making/MapPropertiesMaker.java]: This change isn't 100% clear to me either

Comment (DanVanAtta) [reply game-core/src/main/java/tools/map/making/MapPropertiesMaker.java]: Likely due to unused properties and no longer need the catch, regardless the updates to this file were rolled-back to concentrate on cleanly removing the methods in `MapProperties.java`: e05930636

File: game-core/src/main/java/swinglib/JLabelBuilder.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/main/java/swinglib/JLabelBuilder.java]: I do not quite understand your comment @RoiEXLab. If you're asking why this change, the `truncated` value is always the same as `text`, the 'truncation' feature is unused code.

File: game-core/src/main/java/tools/map/making/MapProperties.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/tools/map/making/MapProperties.java]: I thought so as well, on a double check you are right @ssoloff , here it is:

```
    private String getOutPutString() {
      final StringBuilder outString = new StringBuilder();
      for (final Method outMethod : mapProperties.getClass().getMethods()) {
        final boolean startsWithSet = outMethod.getName().startsWith("out");
        if (!startsWithSet) {
          continue;
        }
        try {
          outString.append(outMethod.invoke(mapProperties));
        } catch (final IllegalArgumentException | InvocationTargetException | IllegalAccessException e) {
          log.log(Level.SEVERE, "Failed to invoke method reflectively: " + outMethod.getName(), e);
        }
      }
      return outString.toString();
    }
  }

```

That is pretty interesting code as only methods with an `out` prefix are used by reflection. Taking this into account, I rolled back the updates to `MapProperties.java` and `*Maker.java`:
- e05930636
- 6acfc5904518e7b760d3481b1f102b51b4c59de7

Then I rolled forward properly removing unused and suppressing unused warnings on 'out*' methods:
- c20b9d5373bae8c1411cf8473250898850ddefa0

File: game-core/src/main/java/swinglib/JLabelBuilder.java (hunk 2 above)
Comment (RoiEXLab) [reply game-core/src/main/java/swinglib/JLabelBuilder.java]: Ah I missed that `maxTextLength` was `Integer.MAX_VALUE`

File: game-core/src/main/java/tools/map/making/MapProperties.java [hunk 4]
```diff
@@ -78,10 +78,7 @@ public void writePropertiesToObject(final List<MapPropertyWrapper<?>> properties
     return colorMap;
   }
 
-  public void setColorMap(final Map<String, Color> value) {
-    colorMap = value;
-  }
-
+  @SuppressWarnings("unused")
```
Comment (RoiEXLab) [inline game-core/src/main/java/tools/map/making/MapProperties.java]: You should probably add a comment somewhere to indicate this is invoked via reflection

File: game-core/src/main/java/tools/map/making/MapProperties.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/tools/map/making/MapProperties.java]: I really appreciate the diligence @ssoloff , it is quite exemplary.

Map properties unused methods clearly needs to be done carefully, IMO probably best to leave that for a dedicated PR going forward. I reverted map property updates from this branch: c20b9d5

File: game-core/src/main/java/tools/map/making/MapProperties.java (hunk 4 above)
Comment (DanVanAtta) [reply game-core/src/main/java/tools/map/making/MapProperties.java]: I agree, this leads to a problem I've been thinking about of how to document many suppression annotations. It's more an issue when the annotations are in many files compared to just one. 

In this case I would think that there should be a comment on how the reflection works and that it is there. I'm not sure if there is a good way to tie each suppression to the class comment to indicate that, or if it's simply better for each and every suppression to write it as:
```
@SuppressWarnigns("unused") // reflective-access
```

I kinda like that in some ways, the comment seems to be a hack though. Perhaps we should consider creating annotations to track these kinds of 'meta-comments'? For example:
```
@AccessedByReflection
@SuppressWarnigns("unused")
```

Any thoughts?

Comment (DanVanAtta) [reply game-core/src/main/java/tools/map/making/MapProperties.java]: (updates to this file were reverted:  https://github.com/triplea-game/triplea/pull/4609/commits/9349c6f3fadee33ffc5182622403f99adf7e1b27)

File: game-core/src/main/java/tools/map/making/MapProperties.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/tools/map/making/MapProperties.java]: Post-revert:
screenshot from 2019-01-25 07-59-16

**Pull request merged**